sql/row: TestExternalRowDataDistSQL failed

sql/row.TestExternalRowDataDistSQL [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17480472?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17480472?buildTab=artifacts#/) on release-24.3 @ [0ea9c22c86975c3e285094f99e1f764f58ef7c6d](https://github.com/cockroachdb/cockroach/commits/0ea9c22c86975c3e285094f99e1f764f58ef7c6d):


```
=== RUN   TestExternalRowDataDistSQL
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/9db56821b86748e0fc537f3d515778f0/logTestExternalRowDataDistSQL2683334059
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    external_row_data_test.go:148: error executing query="ALTER TABLE t RELOCATE VALUES (ARRAY[3], 1), (ARRAY[4], 3), (ARRAY[5], 5)" args=[]: pq: ba: AdminRelocateRange [/Tenant/10/Table/104/1/1/0] RPC error: rpc error: code = Unauthenticated desc = client tenant does not have capability "can_admin_relocate_range" (*kvpb.AdminRelocateRangeRequest)
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9db56821b86748e0fc537f3d515778f0/logTestExternalRowDataDistSQL2683334059
--- FAIL: TestExternalRowDataDistSQL (6.22s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExternalRowDataDistSQL.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-43703

--- a/pkg/sql/row/external_row_data_test.go
+++ b/pkg/sql/row/external_row_data_test.go
@@ -145,7 +145,9 @@ func TestExternalRowDataDistSQL(t *testing.T) {
 
 	// Place leaseholders on nodes 3, 4, 5.
 	r0.Exec(t, `ALTER TABLE t SPLIT AT VALUES (2), (4)`)
-	r0.Exec(t, `ALTER TABLE t RELOCATE VALUES (ARRAY[3], 1), (ARRAY[4], 3), (ARRAY[5], 5)`)
+	r0.ExecSucceedsSoon(
+		t, `ALTER TABLE t RELOCATE VALUES (ARRAY[3], 1), (ARRAY[4], 3), (ARRAY[5], 5)`,
+	)
 
 	asOf := hlc.Timestamp{WallTime: timeutil.Now().UnixNano()}
 

